### PR TITLE
Update zlib-ng to 2.2.1

### PR DESCRIPTION
**Pull request by FantasqueX**
Release: https://github.com/zlib-ng/zlib-ng/releases/tag/2.2.1
ARM diagnostics patch: https://github.com/zlib-ng/zlib-ng/pull/1774

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
zlib-ng also contains branch for s390. It's missing in the patch. Also there are some new run-time arch detection options in the original cmake. I'm looking on it, if it makes sense to integrate them somehow.

**Comment by FantasqueX** [comment]
> zlib-ng also contains branch for s390. It's missing in the patch.

This is intended. There is no s390 recolonization in opencv's processor detection. So, I think s390 part of zlib-ng is not needed.

**Comment by asmorkalov** [comment]
@FantasqueX Thanks a lot for the contribution. I switched to the original project CMake to ease further updates and fixed arm diagnostics. Let's wait for CI.

File: CMakeLists.txt [hunk 1]
```diff
@@ -465,7 +465,7 @@ OCV_OPTION(WITH_CANN "Include CANN support" OFF
 OCV_OPTION(WITH_FLATBUFFERS "Include Flatbuffers support (required by DNN/TFLite importer)" ON
   VISIBLE_IF TRUE
   VERIFY HAVE_FLATBUFFERS)
-OCV_OPTION(WITH_ZLIB_NG "Use zlib-ng instead of zlib" OFF
```
Comment (opencv-alalek) [inline CMakeLists.txt:468]: `OPENCV_FORCE_3RDPARTY_BUILD` should be enabled for testing.

**Comment by asmorkalov** [comment]
@mshabunin @opencv-alalek could you take a look on the PR? I do not know how to handle iOS build failure (internal compiler error). Other platforms are good.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**